// @workos-inc/node/lib/index.worker.mjs
import { A as ApiKeyRequiredException, C as SignatureVerificationException, D as NoApiKeyProvidedException, E as NotFoundException, M as SubtleCryptoProvider, O as BadRequestException, S as UnauthorizedException, T as OauthException, _ as AutoPaginatable, a as OrganizationDomainVerificationStrategy, b as Actions, d as CheckResult, f as CheckOp, g as AuthenticateWithSessionCookieFailureReason, h as serializeRevokeSessionOptions, i as OrganizationDomainState, j as FetchHttpClient, k as GenericServerException, l as WarrantOp, m as RefreshSessionFailureReason, n as ConnectionType, o as DomainDataState, r as GeneratePortalLinkIntent, s as WorkOS, t as createWorkOS, u as ResourceOp, v as PKCE, w as RateLimitExceededException, x as UnprocessableEntityException, y as Webhooks } from "./factory-DClLn4DC.mjs";
//#region src/index.worker.ts
var WorkOSWorker = class extends WorkOS {
	/** @override */
	createHttpClient(options, userAgent) {
		const headers = { "User-Agent": userAgent };
		const configHeaders = options.config?.headers;
		if (configHeaders && typeof configHeaders === "object" && !Array.isArray(configHeaders) && !(configHeaders instanceof Headers)) Object.assign(headers, configHeaders);
		if (this.key) headers["Authorization"] = `Bearer ${this.key}`;
		return new FetchHttpClient(this.baseURL, {
			...options.config,
			headers
		});
	}
	/** @override */
	createWebhookClient() {
		return new Webhooks(new SubtleCryptoProvider());
	}
	getCryptoProvider() {
		return new SubtleCryptoProvider();
	}
	/** @override */
	createActionsClient() {
		return new Actions(new SubtleCryptoProvider());
	}
	/** @override */
	emitWarning(warning) {
		return console.warn(`WorkOS: ${warning}`);
	}
};
//#endregion
export { ApiKeyRequiredException, AuthenticateWithSessionCookieFailureReason, AutoPaginatable, BadRequestException, CheckOp, CheckResult, ConnectionType, DomainDataState, GeneratePortalLinkIntent, GenericServerException, NoApiKeyProvidedException, NotFoundException, OauthException, OrganizationDomainState, OrganizationDomainVerificationStrategy, PKCE, RateLimitExceededException, RefreshSessionFailureReason, ResourceOp, SignatureVerificationException, UnauthorizedException, UnprocessableEntityException, WarrantOp, WorkOSWorker as WorkOS, createWorkOS, serializeRevokeSessionOptions };
