// @workos-inc/node/lib/index.mjs
import { A as ApiKeyRequiredException, C as SignatureVerificationException, D as NoApiKeyProvidedException, E as NotFoundException, M as SubtleCryptoProvider, O as BadRequestException, S as UnauthorizedException, T as OauthException, _ as AutoPaginatable, a as OrganizationDomainVerificationStrategy, b as Actions, c as FeatureFlagsRuntimeClient, d as CheckResult, f as CheckOp, g as AuthenticateWithSessionCookieFailureReason, h as serializeRevokeSessionOptions, i as OrganizationDomainState, j as FetchHttpClient, k as GenericServerException, l as WarrantOp, m as RefreshSessionFailureReason, n as ConnectionType, o as DomainDataState, p as CookieSession, r as GeneratePortalLinkIntent, s as WorkOS, t as createWorkOS, u as ResourceOp, v as PKCE, w as RateLimitExceededException, x as UnprocessableEntityException, y as Webhooks } from "./factory-DClLn4DC.mjs";
//#region src/index.ts
var WorkOSNode = class extends WorkOS {
	/** @override */
	createHttpClient(options, userAgent) {
		const headers = {};
		const configHeaders = options.config?.headers;
		if (configHeaders) if (configHeaders instanceof Headers) configHeaders.forEach((v, k) => headers[k] = v);
		else if (Array.isArray(configHeaders)) configHeaders.forEach(([k, v]) => headers[k] = v);
		else Object.assign(headers, configHeaders);
		headers["User-Agent"] = userAgent;
		if (this.key) headers["Authorization"] = `Bearer ${this.key}`;
		const opts = {
			...options.config,
			timeout: options.timeout,
			headers
		};
		return new FetchHttpClient(this.baseURL, opts, options.fetchFn);
	}
	/** @override */
	createWebhookClient() {
		return new Webhooks(this.getCryptoProvider());
	}
	getCryptoProvider() {
		return new SubtleCryptoProvider();
	}
	/** @override */
	createActionsClient() {
		return new Actions(this.getCryptoProvider());
	}
	/** @override */
	emitWarning(warning) {
		return process.emitWarning(warning, "WorkOS");
	}
};
//#endregion
export { ApiKeyRequiredException, AuthenticateWithSessionCookieFailureReason, AutoPaginatable, BadRequestException, CheckOp, CheckResult, ConnectionType, CookieSession, DomainDataState, FeatureFlagsRuntimeClient, GeneratePortalLinkIntent, GenericServerException, NoApiKeyProvidedException, NotFoundException, OauthException, OrganizationDomainState, OrganizationDomainVerificationStrategy, PKCE, RateLimitExceededException, RefreshSessionFailureReason, ResourceOp, SignatureVerificationException, UnauthorizedException, UnprocessableEntityException, WarrantOp, WorkOSNode as WorkOS, createWorkOS, serializeRevokeSessionOptions };
